### ISSUE TITLE
Incorrect nullable warnings in local functions' closures

### ISSUE THREAD

**Issue by TessenR** [dotnet/roslyn#31423]
**Version Used**: 

Branch C# 8.0: Nullable reference type (29 Oct 2018)
https://github.com/dotnet/csharplang/blob/master/proposals/nullable-reference-types.md
Latest commit 2f8fef by AlekseyTs:
Compare Nullable modifiers while comparing type symbols by default. (#30770)

**Steps to Reproduce**:

Compile the following code:
```cs
#nullable enable
public class C {
    void M(C? c)
    {
        c = new C();
        Local();
        void Local() => c.ToString();
    }
}
```

https://sharplab.io/#v2:EYLgZgpghgLgrgJwgZwLQDk4BstWFiAJQkiQDsBjCAFQE8AHFAGhhAEssAfAYjO13wQABBDJ4CAWABQAAQDMQmQCYhAYSEBvaUJ2KALEICyAClUB+IRQCU23Vqm7HloQF4hZCAHc1xqwG5bJyEAGQB7CigsXwCHIP0Q8MjfVwA+SwA6alCAZRgENjIAc2jAoQBfaTKgA

**Expected Behavior**:

No warnings

**Actual Behavior**:

`warning CS8602: Possible dereference of a null reference.`

**Comment by TessenR** [comment]
Similarly non-null state in local function is not propagated through its invocation

```cs
#nullable enable
public class C {
    void M(C? c)
    {
      void Local() => c = new C();
      Local();
      c.ToString(); // incorrect warning CS8602: Possible dereference of a null reference.
    }
}
```

**Comment by jaredpar** [comment]
Thanks for filing this. This appears to be a duplicate of #29892.
